doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+| **Default**                                                  | None |
+| **Command mode**                                             | EXEC |
+| **Example**                                                  | Sonic# show running-configuration pim<br>!<br>interface vlan1001<br>ip pim sparse-mode<br>!<br>interface vlan2001<br>ip pim sparse-mode<br>!<br><br>Sonic# |
+| **Support**                                                  | 4.1 |
+
+**Note:**  
+The FRR PIM stack is enhanced to provide the support for the newly added command "Show ip pim rp < grp-addr >" and json support is added for the existing "show ip pim assert" command.
+   
+#### 3.5.2.4 Debug Commands
+No new debug commands are added as part of PIM-ASM implementation.
+#### 3.5.2.5 IS-CLI Compliance
+NA
+
+### 3.5.3 REST API Support
+Rest API support is there for all the existing fields and the ones that are newly added for PIM-SM like RP and spt-switchover. The sample curl commands for CRUD on RP are given below
+
+1) curl -X POST "https://100.104.98.88/restconf/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=PIM,pim/pim" -H "accept: */*" -H "Authorization: Basic YWRtaW46YnJvYWRjb20=" -H "Content-Type: application/yang-data+json" -d "{\"openconfig-network-instance:global\":{\"rendezvous-points\":{\"rendezvous-point\":[{\"address\":\"4.4.4.4\"}]}}}"
+2) curl -X PUT "https://100.104.98.88/restconf/data/openconfig-network-instance:network-instances/network-instance=Vrf_green/protocols/protocol=PIM,pim/pim/global" -H "accept: */*" -H "Authorization: Basic YWRtaW46YnJvYWRjb20=" -H "Content-Type: application/yang-data+json" -d "{\"openconfig-network-instance:global\":{\"rendezvous-points\":{\"rendezvous-point\":[{\"address\":\"4.4.4.4\",\"config\":{\"address\":\"4.4.4.4\",\"multicast-groups\":\"P1\"}}]}}}"
+3) curl -X PATCH "https://100.104.98.88/restconf/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=PIM,pim/pim/global" -H "accept: */*" -H "Authorization: Basic YWRtaW46YnJvYWRjb20=" -H "Content-Type: application/yang-data+json" -d "{\"openconfig-network-instance:global\":{\"rendezvous-points\":{\"rendezvous-point\":[{\"address\":\"4.4.4.4\",\"config\":{\"address\":\"4.4.4.4\",\"multicast-groups\":\"P1\"}}]}}}"
+4) curl -X GET "https://100.104.98.88/restconf/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=PIM,pim/pim/global" -H "accept: application/yang-data+json" -H "Authorization: Basic YWRtaW46YnJvYWRjb20="
+5) curl -X DELETE "https://100.104.98.88/restconf/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=PIM,pim/pim/global" -H "accept: */*" -H "Authorization: Basic YWRtaW46YnJvYWRjb20="
+
+### 3.5.4 gNMI Support
+
+
+# 4 Flow Diagrams
+
+## 4.1 PIM Config Flow
+![](pim_images/PIM_STATIC_RP_CONFIG_FLOW.png)
+
+## 4.2 PIM Data packet Flow
+![](2021-12-06-13-10-56.png)
```

Comment:
@vidyac86 : Please check, the images for 4.2, 4.3 & 4.4 are missing. Also we had one flow diagram called "PIM KAT MFPM channel error handling". Please add that.

Reply:
I have added them